frontend/src/services/ApiService.ts

```diff
@@ -0,0 +1,59 @@
+import axios, { AxiosError } from 'axios'
+
+import { LOCAL_STORAGE_EVENT, LOGGED_IN_KEY } from '~constants/localStorage'
+
+const API_BASE_URL = process.env.REACT_APP_BASE_URL ?? '/api/v3'
+export class HttpError extends Error {
+  code: number
+  constructor(message: string, code: number) {
+    super(message)
+    this.code = code
+  }
+}
+
+/**
+ * Converts possible AxiosError objects to normal Error objects
+ *
+ * @returns HttpError if AxiosError, else original error
+ */
+export const transformAxiosError = (e: Error): HttpError | Error => {
+  if (axios.isAxiosError(e) && e.response) {
+    const statusCode = e.response.status
+    if (statusCode === 429) {
+      return new HttpError('Please try again later.', statusCode)
+    }
+    if (typeof e.response.data === 'string') {
+      return new HttpError(e.response.data, statusCode)
+    }
+    if (e.response.data?.message) {
+      return new HttpError(e.response.data.message, statusCode)
+    }
+    if (e.response.statusText) {
+      return new HttpError(e.response.statusText, statusCode)
+    }
```

Comment:
I'm not sure we want to do this - the status text ("Forbidden", "Unauthorized") is rarely useful info for a user. personally I'd rather default to a generic "contact us" error message at this point

Reply:
Out of scope of this PR? (even though I created it here :X) We can revisit this in the future?